google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -482,12 +482,19 @@ def delete table, *id
         #     end
         #   end
         #
-        def transaction
+        def transaction &block
           ensure_service!
           tx_session = session
           tx_grpc = @project.service.begin_transaction tx_session.path
           tx = Transaction.from_grpc(tx_grpc, tx_session)
-          yield tx
+          begin
+            block.call tx
+          rescue Google::Cloud::AbortedError
+            # Retry the entire transaction
```

Comment:
I believe this should be performed with a backoff delay, using a value returned by the service?